### PR TITLE
Separate user's and root's commands

### LINKED ISSUE #62: Enable roxy to be used as a library

### PR DESCRIPTION
**Pull request by sehkone**
The functions listed in `lib.rs` are not covering all of required commands. The rest will be added.

Closes #62

### CODE REVIEW

File: src/common.rs [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    pub arg: Vec<u8>,
+}
+
+impl NodeRequest {
+    /// # Arguments
+    /// * cmd<T>: command arguments. T: type of arguments
+    ///
+    /// # Errors
+    /// * fail to serialize command
+    #[allow(dead_code)]
```
Comment (msk) [inline src/common.rs:36]: Any plan to use this?

File: src/common.rs [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        T: Serialize,
+    {
+        match bincode::serialize(&cmd) {
+            Ok(arg) => Ok(NodeRequest { kind, arg }),
+            Err(e) => Err(anyhow!("Error: {}", e)),
+        }
+    }
+
+    #[allow(dead_code)]
+    pub fn debug<T>(&self)
```
Comment (msk) [inline src/common.rs:48]: Isn't it better to `impl Debug`?

File: src/common.rs [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct NicOutput {
+    pub addresses: Option<Vec<String>>,
+    pub dhcp4: Option<bool>,
+    pub gateway4: Option<String>,
+    pub nameservers: Option<Vec<String>>,
+}
+
+impl NicOutput {
+    #[allow(dead_code)]
```
Comment (msk) [inline src/common.rs:126]: Please add a comment explaining why this shouldn't be deleted. `#[allow(dead_code)]` must be treated in the same way as commented-out codes.

File: src/common.rs [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        NicOutput {
+            addresses: nic.addresses.clone(),
+            dhcp4: nic.dhcp4,
+            gateway4: nic.gateway4.clone(),
+            nameservers,
+        }
+    }
+}
+
+/// Run linux command and return it's output
```
Comment (msk) [inline src/common.rs:179]: A rustdoc comment starts with "Runs ..." rather than "Run ...".

**Review by msk — CHANGES_REQUESTED** [review]

File: src/common.rs (hunk 1 above)
Comment (sehkone) [reply src/common.rs]: Corrected

File: src/common.rs (hunk 2 above)
Comment (sehkone) [reply src/common.rs]: Erased because it is not used

File: src/common.rs (hunk 3 above)
Comment (sehkone) [reply src/common.rs]: Corrected

File: src/common.rs (hunk 4 above)
Comment (sehkone) [reply src/common.rs]: I corrected this but I didn't check others which may exist. Whenever I find similar ones, I will correct them.

File: src/root.rs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+mod sshd;
+mod syslog;
+pub(crate) mod task;
+mod ufw;
+
+use super::common::{run_command_output, Nic, NicOutput, SubCommand, DEFAULT_PATH_ENV};
+use anyhow::{anyhow, Result};
+use std::process::Command;
+
+/// Run linux command
```
Comment (msk) [inline src/root.rs:14]: Run → Runs

File: src/root/hwinfo.rs [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+use anyhow::{anyhow, Result};
+use std::{
+    fs::{self, OpenOptions},
+    io::Write as IoWrite,
+};
+
+// TODO: should change this path to /usr/local/aice/conf/version?
+const DEFAULT_VERSION_PATH: &str = "/etc/version";
+
+/// Set OS or Product version. Refer /etc/version
```
Comment (msk) [inline src/root/hwinfo.rs:11]: Set → Sets

**Review by msk — CHANGES_REQUESTED** [review]

File: src/root.rs (hunk 5 above)
Comment (sehkone) [reply src/root.rs:14]: There are many mistakes like this here and there. This should be corrected properly by the author. I will post an issue.

Comment (sehkone) [reply src/root.rs:14]: I post an issue #69

**Review by msk — APPROVED** [review]

**Pull request merged**